firmware/baseband/proc_btlerx.hpp

```diff
@@ -42,48 +43,1931 @@ class BTLERxProcessor : public BasebandProcessor {
     void on_message(const Message* const message) override;
 
    private:
+#define SAMPLE_PER_SYMBOL 1
+#define LEN_DEMOD_BUF_ACCESS 32
+#define DEFAULT_ACCESS_ADDR 0x8E89BED6
+#define NUM_ACCESS_ADDR_BYTE 4
+
+    std::array<std::string, 16> ADV_PDU_TYPE_STR{
+        "ADV_IND",
+        "ADV_DIRECT_IND",
+        "ADV_NONCONN_IND",
+        "SCAN_REQ",
+        "SCAN_RSP",
+        "CONNECT_REQ",
+        "ADV_SCAN_IND",
+        "RESERVED0",
+        "RESERVED1",
+        "RESERVED2",
+        "RESERVED3",
+        "RESERVED4",
+        "RESERVED5",
+        "RESERVED6",
+        "RESERVED7",
+        "RESERVED8"};
+
+    typedef enum {
+        ADV_IND = 0,
+        ADV_DIRECT_IND = 1,
+        ADV_NONCONN_IND = 2,
+        SCAN_REQ = 3,
+        SCAN_RSP = 4,
+        CONNECT_REQ = 5,
+        ADV_SCAN_IND = 6,
+        RESERVED0 = 7,
+        RESERVED1 = 8,
+        RESERVED2 = 9,
+        RESERVED3 = 10,
+        RESERVED4 = 11,
+        RESERVED5 = 12,
+        RESERVED6 = 13,
+        RESERVED7 = 14,
+        RESERVED8 = 15
+    } ADV_PDU_TYPE;
+
+    uint8_t macAddress[6];
+    int checksumReceived = 0;
+
+    typedef struct
+    {
+        uint8_t Data[31];
+    } ADV_PDU_PAYLOAD_TYPE_0_2_4_6;
+
+    typedef struct
+    {
+        uint8_t A1[6];
+    } ADV_PDU_PAYLOAD_TYPE_1_3;
+
+    typedef struct
+    {
+        uint8_t AdvA[6];
+        uint8_t AA[4];
+        uint32_t CRCInit;
+        uint8_t WinSize;
+        uint16_t WinOffset;
+        uint16_t Interval;
+        uint16_t Latency;
+        uint16_t Timeout;
+        uint8_t ChM[5];
+        uint8_t Hop;
+        uint8_t SCA;
+    } ADV_PDU_PAYLOAD_TYPE_5;
+
+    typedef struct
+    {
+        uint8_t payload_byte[40];
+    } ADV_PDU_PAYLOAD_TYPE_R;
+
     static constexpr size_t baseband_fs = 4000000;
     static constexpr size_t audio_fs = baseband_fs / 8 / 8 / 2;
 
-    std::array<complex16_t, 512> dst{};
+    uint_fast32_t crc_update(uint_fast32_t crc, const void* data, size_t data_len);
+    uint_fast32_t crc24_byte(uint8_t* byte_in, int num_byte, uint32_t init_hex);
+    bool crc_check(uint8_t* tmp_byte, int body_len, uint32_t crc_init);
+    uint32_t crc_init_reorder(uint32_t crc_init);
+
+    uint32_t crc_initalVale = 0x555555;
+    uint32_t crc_init_internal = 0x00;
+
+    void scramble_byte(uint8_t* byte_in, int num_byte, const uint8_t* scramble_table_byte, uint8_t* byte_out);
+    // void demod_byte(int num_byte, uint8_t *out_byte);
+    int parse_adv_pdu_payload_byte(uint8_t* payload_byte, int num_payload_byte, ADV_PDU_TYPE pdu_type, void* adv_pdu_payload);
+
+    std::array<complex16_t, 1024> dst{};
```

Comment:
Not really something to change, but something to keep in mind -- the RAM working area for baseband processors isn't very large and this class is quite large with all of its buffers.